languages/sqlite/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+
+	"github.com/kr/pty"
+	"golang.org/x/sys/unix"
+
+	"github.com/replit/prybar/utils"
+)
+
+var Instance = &SQLite{}
+
+type SQLite struct {}
+
+// constructConfigFile generates commands to configure the sqlite CLI.
+// It writes them to a temporary file and returns its pathname.
+func constructConfigFile(config *utils.Config) string {
+	f, err := ioutil.TempFile("", "sqlite-config")
+	if err != nil {
+		panic(err)
+	}
+	defer f.Close()
+
+	// main and continuation prompts
+	// TODO: this probably doesn't handle quotation marks properly
+	f.WriteString(fmt.Sprintf(".prompt '%s' '%s'\n", config.Ps1, config.Ps2))
+
+	return f.Name()
+}
+
+func disableEcho(file *os.File) {
+	fd := int(file.Fd())
+	termios, err := unix.IoctlGetTermios(fd, unix.TCGETS)
+	if err != nil {
+		panic(err)
+	}
+
+	termios.Lflag &^= unix.ECHO
+
+	if err := unix.IoctlSetTermios(fd, unix.TCSETS, termios); err != nil {
+		panic(err)
+	}
+}
+
+func Execute(config *utils.Config) {
+	configFile := constructConfigFile(config)
+	args := []string{"-init", configFile}
+	cmd := exec.Command("sqlite3", args...)
+
+	ptty, tty, err := pty.Open()
+	if err != nil {
+		panic(err)
+	}
+	disableEcho(ptty)
+
+	cmd.Stderr = tty
+	cmd.Stdin = tty
+	cmd.Stdout = tty
+
+	err = cmd.Start()
+	if err != nil {
+		panic(err)
+	}
+
+	// file to execute
+	if len(config.Args) > 1 {
+		panic("too many arguments")
+	}
+	if len(config.Args) == 1 {
+		fileToRun := config.Args[0]
+		ptty.WriteString(fmt.Sprintf(".read %s\n", fileToRun))
+	}
+	
+	// set up I/O
+	go io.Copy(os.Stderr, ptty)
+	if config.Quiet {
+		go io.Copy(os.Stdout, filter(ptty))
+	} else {
+		go io.Copy(os.Stdout, ptty)
+	}
+	io.Copy(ptty, os.Stdin)
+}
+
+// filter removes all output until we get a prompt
+func filter(src io.Reader) io.Reader {
+	scanner := bufio.NewScanner(src)
+	for scanner.Scan() {
+		if scanner.Text() == "Enter \".help\" for usage hints." {
```

Comment:
this seems super flimsy. Can we find a better way that doesn't require finding strings in the output.

Reply:
Yes, I agree